pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -139,6 +140,84 @@ type PerconaXtraDBCluster struct {
 	Status PerconaXtraDBClusterStatus `json:"status,omitempty"`
 }
 
+func (cr *PerconaXtraDBCluster) Validate() error {
+	if len(cr.Name) > clusterNameMaxLen {
+		return errors.Errorf("cluster name (%s) too long, must be no more than %d characters", cr.Name, clusterNameMaxLen)
+	}
+
+	c := cr.Spec
+
+	if c.PXC == nil {
+		return errors.Errorf("spec.pxc section is not specified. Please check %s cluster settings", cr.Name)
+	}
+
+	if c.PXC.Image == "" {
+		return errors.New("pxc.Image can't be empty")
+	}
+
+	if c.PXC.VolumeSpec == nil {
+		return errors.New("PXC: volumeSpec should be specified")
+	}
+
+	if err := c.PXC.VolumeSpec.validate(); err != nil {
+		return errors.Wrap(err, "PXC: validate volume spec")
+	}
+
+	if c.HAProxy != nil && c.HAProxy.Enabled &&
+		c.ProxySQL != nil && c.ProxySQL.Enabled {
+		return errors.New("can't enable both HAProxy and ProxySQL please only select one of them")
+	}
+
+	if c.HAProxy != nil && c.HAProxy.Enabled {
+		if c.HAProxy.Image == "" {
```

Comment:
Can be a part of previous `if` statement

Reply:
It's in the same style as proxysql and etc.
Plus, if we add more checks, we will need to rewrite it